metadata/metadata.go

```diff
@@ -54,6 +54,70 @@ func AddLabels(obj v1.Object, entries map[string]string) error {
 	return nil
 }
 
+// CopyAnnotationsByPrefix copies all annotations from a source object to a destination object where the key matches
+// the specified sourcePrefix.
+func CopyAnnotationsByPrefix(source, destination v1.Object, prefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+	if source.GetAnnotations() == nil {
+		return nil
+	}
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyByPrefix(source.GetAnnotations(), destination.GetAnnotations(), prefix)
+	return nil
+}
+
+// CopyAnnotationsWithPrefixReplacement copies all annotations from a source object to a destination object where the
+// key matches the specified sourcePrefix.
+// The source prefix will be replaced with the destination prefix.
+func CopyAnnotationsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+	if source.GetAnnotations() == nil {
+		return nil
+	}
+	if destination.GetAnnotations() == nil {
+		destination.SetAnnotations(make(map[string]string))
+	}
+	copyWithPrefixReplacement(source.GetAnnotations(), destination.GetAnnotations(), sourcePrefix, destinationPrefix)
+	return nil
+}
+
+// CopyLabelsWithPrefixReplacement copies all labels from a source object to a destination object where the key matches the specified sourcePrefix.
+// If destinationPrefix is different from sourcePrefix, the sourcePrefix will be replaced while performing the copy.
+func CopyLabelsWithPrefixReplacement(source, destination v1.Object, sourcePrefix, destinationPrefix string) error {
+	if source == nil || destination == nil {
+		return errors.New("object cannot be nil")
+	}
+	if source.GetLabels() == nil {
+		return nil
+	}
+	if destination.GetLabels() == nil {
+		destination.SetLabels(make(map[string]string))
+	}
+	copyWithPrefixReplacement(source.GetLabels(), destination.GetLabels(), sourcePrefix, destinationPrefix)
+	return nil
+}
+
+// CopyLabelsByPrefix copies all labels from a source object to a destination object where the key matches the specified sourcePrefix.
+func CopyLabelsByPrefix(source, destination v1.Object, prefix string) error {
```

Comment:
This function is not in alphabetical order. Move it one up.

Reply:
Updated, thanks!